pkg/apis/nginx/v1alpha1/nginx_types.go

```diff
@@ -59,20 +59,27 @@ type Nginx struct {
 	Status NginxStatus `json:"status,omitempty"`
 }
 
-type NginxPod struct {
+type PodStatus struct {
 	// Name is the name of the POD running nginx
 	Name string `json:"name"`
 	// PodIP is the IP if the POD
 	PodIP string `json:"podIP"`
 }
 
-type NginxService struct {
-	// Name is the name of the service in front of the nginx
+type ServiceStatus struct {
+	// Name is the name of the Service created by nginx
 	Name string `json:"name"`
+}
+
+type NginxService struct {
 	// Type is the type of the service
 	Type string `json:"type"`
-	// ServiceIP is the IP of the service
-	ServiceIP string `json:"serviceIP"`
+	// LoadBalancerIP is an optional load balancer IP for the service
+	LoadBalancerIP string `json:"serviceIP"`
```

Comment:
The JSON annotation should be `loadbalancerIP`, right?

```suggestion
	LoadBalancerIP string `json:"loadbalancerIP"`
```

Reply:
Ouch.